### PR TITLE
Add gzip compression option

### PR DESCRIPTION
**Pull request by o1egl**
https://deliveroo.atlassian.net/browse/CAI-923
https://deliveroo.atlassian.net/browse/CAI-878

### CODE REVIEW

File: jsonrest.go [hunk 1]
```diff
@@ -12,9 +12,15 @@ import (
 	"strings"
 	"sync"
 
+	"github.com/NYTimes/gziphandler"
```
Comment (thoeni) [inline jsonrest.go:15]: I remember this one is what @johngibb once shared with me!

**Review by thoeni — COMMENTED** [review]
This seems to just add the option but doesn't use it?

**Comment by o1egl** [comment]
> This seems to just add the option but doesn't [use it](https://deliveroo.atlassian.net/browse/CAI-878)?

Not sure I'm following you. It is used [here](https://github.com/deliveroo/jsonrest-go/pull/23/files#diff-6d1f6da624fcc409815411c88d49d64121e64c396c463591a6f72deb1ed22fb2R288-R292), and there are tests that actually check body compression. We also need to bump the version of this lib and make corresponding changes in the rs-menus-api after merging this PR

**Review by thoeni — APPROVED** [review]

File: jsonrest.go [hunk 2]
```diff
@@ -12,9 +12,15 @@ import (
...
 	"github.com/julienschmidt/httprouter"
 )
 
+const (
+	HeaderAcceptEncoding = "Accept-Encoding"
+	GzipEncoding         = "gzip"
```
Comment (raycchan) [inline jsonrest.go:21]: what is this magic? I don't see these consts used anywhere, nor used in the `rs-menus-api` implementation, but clearly your tests here show that it works 🪄 ✨

**Pull request merged**